sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1077,7 +1080,8 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       case expressions =>
         expressions
     }
-    val function = UnresolvedFunction(visitFunctionName(ctx.qualifiedName), arguments, isDistinct)
+    val function = UnresolvedFunction(visitFunctionName(ctx.qualifiedName, trimFuncName),
```

Comment:
I see, you want to combine all together. 
but what do you think this approach. I am thinking in the future, if there is another function need to have special handling, this way may be easy to do? 
`  val function = if (!trimFuncName.isDefined) {
      UnresolvedFunction(visitFunctionName(ctx.qualifiedName), arguments, isDistinct)
    } else {
      UnresolvedFunction(visitTrimFunctionName(ctx.qualifiedName, trimFuncName.get),
        arguments, isDistinct)
    }
`
` protected def visitTrimFunctionName(
      ctx: QualifiedNameContext,
      trimFuncName: String): FunctionIdentifier = {
    ctx.identifier().asScala.map(_.getText) match {
      case Seq(db, fn) => FunctionIdentifier(trimFuncName, Option(db))
      case Seq(fn) => FunctionIdentifier(trimFuncName, None)
      case other => throw new ParseException(s"Unsupported function name '${ctx.getText}'", ctx)
    }
  }`